db/migrate/20240801094244_populate_everyone_role.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class PopulateEveryoneRole < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  class UserRole < ApplicationRecord
+    EVERYONE_ROLE_ID = -99
+
+    FLAGS = {
+      invite_users: (1 << 16),
+    }.freeze
+  end
+
+  def up
+    UserRole.create!(id: UserRole::EVERYONE_ROLE_ID, permissions: UserRole::FLAGS[:invite_users])
```

Comment:
I feel like I'm missing something obvious, but -- shouldnt this role already exist from the seeds?

Reply:
Interesting ... I'd never noticed that migration before. So currently if someone runs `bin/rails db:migrate:setup` (or some variation of "run migrations and seeds" via whatever commands you might use), they wind up loading all the seeds once, and then the roles seed specifically a second time during that one migration. This is basically harmless since the seed files are all idempotent and will just not do anything after the first run.

I assume the reason for this migration then is that if we're going to keep the FK constraint and also have a default value, we need to be sure the row exists _before_ the migration adding the default is run, and thus can't rely on a post-migrations-run or post-schema-load db seed load?

Reply:
Yes, that's it!

Though, once again, taking a step back, I would be happy with a migration-less approach that may do a N+1 once if the default role doesn't exist yet at runtime, as long as we can avoid N+1s when the role does exist.

Reply:
Yeah, agreed on that ... I think we safely can assume that during app runtime all migrations have been run and the seeds have been loaded? So for the purposes of "try to stop the N+1 on users with null role values", we can assume that the everyone user role (since loaded by seeds) does exist?

If we CANT assume that ... that's rough. Maybe an initializer or something similar to the RAILS_ENV check on boot that would just abort/warn if the record is not there (with a message to run the seeds?).

If we can assume that, I think that path of trying to mitigate with after_initialize or some other migration-less approach is at least worth trying first before resorting to this migration approach.

Reply:
The main issue with `after_initialize` is that it would set the value to `-99` in the database, while we ideally do not want that to be persisted in the database, for index size reasons mainly.

Reply:
This preference (of preferring nil in DB with model level defaults) is sort of similar to https://github.com/mastodon/mastodon/pull/31908 ... if that's truly a team pref in a general sense rather than case by case, might be good to capture that somewhere.

It's also not 100% clear to me whether the DB index issue is the prime driver on the other linked one, or if that's more about "preserve the ability to change defaults for users who have never set something", or something else.